app-core.js

```diff
@@ -155,7 +154,36 @@ async function serveOctoFarmNormally(app, quick_boot = false) {
     const stateRunnerReport = await Runner.init();
     logger.info("OctoFarm State returned", stateRunnerReport);
 
-    await FilamentClean.start();
+    // await FilamentClean.start();
+    // TODO race condition
+    await softwareUpdateChecker.syncLatestOctoFarmRelease(false).then(() => {
+      softwareUpdateChecker.checkReleaseAndLogUpdate();
+    });
+
+    // Hacky database check due to shoddy layout of code...
+    // const mongoose = require("mongoose");
+    const serverSettings = require("./server_src/settings/serverSettings");
+
+    const printersInformation = PrinterClean.listPrintersInformation();
+    await PrinterClean.sortCurrentOperations(printersInformation);
+    await PrinterClean.statisticsStart();
+    await PrinterClean.createPrinterList(
+      printersInformation,
+      serverSettings.filamentManager
+    );
+
+    // let interval = false;
+    // if (interval === false) {
+    //   interval = setInterval(async () => {
```

Comment:
@NotExpectedYet I've not yet introduced a periodic call here. Do we have anything which is more abstract than setInterval?
Is this robust against errors f.e.? And what about suspending/stopping the asynchronous code or managing this during tests? Got any idea for that? (Im not mentioning a background job runner to not immediately force us in that train of thought)

Reply:
Not that I'm aware of. The only way to repeat a function that I'm aware of would be with setInterval or a function call loop and a setTimeout to trigger the next call of the function. A nested setTimeout would be more controllable than a setInterval. 

We could write something that does those functions but I wonder if a library exists.

Reply:
Im thinking about writing a manager to manage the intervals.
1) Request a time interval + pass a sync/async callback
2) Manager saves/memorizes the interval number and can restart it/kill it/idle it